### PR TITLE
Cleanup context in devfile integration tests

### LINKED ISSUE #3801: Clean up context in devfile integration tests
Currently most of the devfile integration tests that use context flag `--context`, change to the context directory in `BeforeEach` even though we're using the full path in `--context`

Ideally, we should not switch to the context directory when using `--context`. Clean up all devfile integration tests for the flag.

/area devfile
/kind cleanup

### PR DESCRIPTION
**Pull request by valaparthvi**
**What type of PR is this?**
/kind cleanup

**What does this PR do / why we need it**:
This PR cleans up code that renders the use of `--context` useless.

**Which issue(s) this PR fixes**:

Fixes #3801

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
Run the tests fixed by the PR.

### CODE REVIEW

**Comment by rnapoles-rh** [comment]
PSI tests are optional at this time

**Review by rnapoles-rh — APPROVED** [review]
/lgtm

**Comment by girishramnani** [comment]
why lgtm a WIP PR @rnapoles-rh ?

**Comment by rnapoles-rh** [comment]
> why lgtm a WIP PR @rnapoles-rh ?

sorry, I missed the tag

File: tests/integration/devfile/cmd_devfile_describe_test.go [hunk 1]
```diff
@@ -28,6 +27,13 @@ var _ = Describe("odo devfile describe command tests", func() {
 	})
 
 	Context("When executing odo describe", func() {
+		JustAfterEach(func() {
+			// odo delete requires changing directory because it does not work as intended with --context
```
Comment (dharmit) [inline tests/integration/devfile/cmd_devfile_describe_test.go:31]: Can you please make this a TODO and add the issue(s) information?

**Review by dharmit — CHANGES_REQUESTED** [review]

**Comment by dharmit** [comment]
@valaparthvi i see a bunch of integration test failures. It's a mess to read through them since the logs are garbled. Can you please take a look if these failures are relevant to the changes you've made in this PR? If not relevant, I think you could trigger a retest.

**Comment by valaparthvi** [comment]
Those failures are related to my changes, I am still working on them. Sorry, I should've put it back to WIP.

**Comment by dharmit** [comment]
> Those failures are related to my changes, I am still working on them. Sorry, I should've put it back to WIP.

Np. You don't really need to put it back to WIP when CI helps you discover that some changes need to be made. I generally put a PR in WIP when I know there's some work pending from my side and I don't want the team to waste time reviewing the PR. YMMV. :)

**Comment by valaparthvi** [comment]
/test v4.7-integration-e2e

> INFO[2021-04-06T14:17:22Z] • Failure [69.413 seconds]      
INFO[2021-04-06T14:17:22Z] odo devfile push command tests               
INFO[2021-04-06T14:17:22Z] /go/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_push_test.go:20 
INFO[2021-04-06T14:17:22Z]   Verify devfile push works                  
INFO[2021-04-06T14:17:22Z]   /go/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_push_test.go:64 
INFO[2021-04-06T14:17:22Z]     should execute the default build and run command groups if present [It] 
INFO[2021-04-06T14:17:22Z]     /go/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_push_test.go:345 
INFO[2021-04-06T14:17:22Z]                                              
INFO[2021-04-06T14:17:22Z]     Unexpected error:                   
INFO[2021-04-06T14:17:22Z]         <*errors.errorString | 0xc0004281b0>: { 
INFO[2021-04-06T14:17:22Z]             s: "cmd [ps -ef] failed with error Signal 23 (URG) caught by ps (3.3.15).\nps:ps/display.c:66: please report this bug\ncommand terminated with exit code 1\n on pod jivyym-744c4797df-2nv7n", 
INFO[2021-04-06T14:17:22Z]         }                                    
INFO[2021-04-06T14:17:22Z]         cmd [ps -ef] failed with error Signal 23 (URG) caught by ps (3.3.15). 
INFO[2021-04-06T14:17:22Z]         ps:ps/display.c:66: please report this bug 
INFO[2021-04-06T14:17:22Z]         command terminated with exit code 1  
INFO[2021-04-06T14:17:22Z]          on pod jivyym-744c4797df-2nv7n      
INFO[2021-04-06T14:17:22Z]     occurred                             
INFO[2021-04-06T14:17:22Z]                                              
INFO[2021-04-06T14:17:22Z]     /go/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_push_test.go:364

Retesting. Although, the failure is unrelated to the changes made by this PR.

**Comment by valaparthvi** [comment]
/test v4.7-integration-e2e

Retesting, because not all tests were run.

**Comment by valaparthvi** [comment]
@rnapoles-rh it's now RFR.

**Review by rnapoles-rh — APPROVED** [review]
Nice cleanup
/lgtm

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -16,7 +16,7 @@
 > /kind documentation
 > [skip ci]
 
-**What does does this PR do / why we need it**:
+**What does this PR do / why we need it**:
```
Comment (dharmit) [inline .github/PULL_REQUEST_TEMPLATE.md:19]: Thanks for fixing this! 👍🏾

**Review by dharmit — COMMENTED** [review]
/approve

**Pull request merged**